gnovm/pkg/gnolang/machine.go

```diff
@@ -481,10 +480,16 @@ func (m *Machine) injectLocOnPanic() {
 // Add files to the package's *FileSet and run them.
 // This will also run each init function encountered.
 func (m *Machine) RunFiles(fns ...*FileNode) {
-	m.runFiles(fns...)
+	m.runFiles(nil, fns...)
+}
+
+// RunFilesWithMemPkg is almost the same as RunFiles; the difference is that it
+// saves the mempackage to the store.
```

Comment:
the name is inconsistent with its behavior though...

Reply:
gut tells me not to add saving logic into runFiles logic, but instead runFiles could return new init functions as []Name, and the caller can save before or after as needed.

I'mstill trying to wrap my head around why it's needed. thinking..